synapse/lib/stormlib/imap.py

```diff
@@ -218,6 +218,17 @@ async def postAnit(self):
 
         return self
 
+    async def fini(self):
+        if self.isfini:
+            return
+
+        try:
```

Comment:
I almost wonder if we should run the logout + super().fini() as a schedCoro off of the Cortex?

This could wait_for could wait up to 5 seconds between the query being "done" and the caller getting a `fini` message.

Reply:
we can tune that down if you think it's too high. I picked 5 seconds randomly.

Reply:
We originally moved it to a background task in #4351 to get the logout action ( which, should be fast) out of line with blocking fini on a external service.

Reply:
updated